hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndexUtils.java

```diff
@@ -636,7 +638,7 @@ public static <R> HoodieData<HoodieRecord<R>> tagGlobalLocationBackToRecords(
             return Pair.of(incomingRecord, Option.empty());
           }
         });
-    return shouldUpdatePartitionPath
+    return shouldUpdatePartitionPath || isMoRTable
```

Comment:
So could the `COMMIT_TIME_ORDERING` also be excluded from using `mergeForPartitionUpdatesIfNeeded`?  Reading existing records from storage and merge with the incoming records adds additional overhead so see if it is possible to exclude cases that do not need the merging.

Also wondering if there is easy way to check there is non-empty `Option<HoodieRecordGlobalLocation>` in `incomingRecordsAndLocations`; then trigger `mergeForPartitionUpdatesIfNeeded` if so.

In the future, once we store ordering value in RLI, for `EVENT_TIME_ORDERING`, the `mergeForPartitionUpdatesIfNeeded` can be skipped and comparison of the ordering value of incoming deletes and that of RLI entry would determine if the RLI entry should be deleted.

Reply:
seems not right, the check there will decide whether to issue retraction deletes for partition updates, we can not force make it true for MOR table regardless of the config `shouldUpdatePartitionPath`, that would incur inconsiscency semantics for COW and MOR.

Reply:
That's what I mean.  Beyond what should go through the merging on master, i.e., `shouldUpdatePartitionPath = true`, additionally, if it's non `COMMIT_TIME_ORDERING` merge mode, the merging of incoming and existing records should also happen.